napari/layers/_labels_layer/view/properties.py

```diff
@@ -2,14 +2,15 @@
 from ..._base_layer import QtLayer
 
 
-class QtImageLayer(QtLayer):
+class QtLabelsLayer(QtLayer):
     def __init__(self, layer):
         super().__init__(layer)
 
         self.layer.events.colormap.connect(self._on_colormap_change)
 
-        self.colormap_update = QPushButton('shuffle colors', parent=self)
+        self.colormap_update = QPushButton('shuffle colors')
         self.colormap_update.clicked.connect(self.changeColor)
+        self.grid_layout.addWidget(self.colormap_update, 3, 0, 1, 2)
```

Comment:
Aaaaaah! 👍 

I'm confused why you don't need parent?

Reply:
Wait: what are those numbers?

Reply:
actually the `parent` thing is probably a good idea - helps with ensuring the widget gets deleted if the parent gets deleted. Not strictly necessary, but good to have, and wasn't the source of the button being in the wrong place. We should probably go through and add parents to all our widgets, but can do that in a new PR. Probably worth making an issue about.

Those numbers are slightly horribly hard coded values that determine where in the grid the new widget is getting added row 3, column 0, row_span 1, column span 2. The 3 in that list should probably be calculated by looking at the number of rows already present and adding 1. This problem is present in all of our `properties` code and should again probably be fixed across the board

Reply:
This has been raised in issue #185